rir/src/runtime/Assumptions.cpp

```diff
@@ -55,7 +58,7 @@ std::ostream& operator<<(std::ostream& out, const Assumptions& a) {
     return out;
 }
 
-constexpr std::array<Assumption, 3> Assumptions::ObjAssumptions;
-constexpr std::array<Assumption, 3> Assumptions::EagerAssumptions;
+constexpr std::array<Assumption, 5> Assumptions::ObjAssumptions;
+constexpr std::array<Assumption, 4> Assumptions::EagerAssumptions;
```

Comment:
Why do we now have 5 and 4, when before it was 3 and 3?

Reply:
just wanted to see who comments first on it :)

Reply:
before, there was one flag for all are eager + specific flags for 1-3 when the fomer was not set (same for obj). but that was just to brittle. so i removed the all are eager flag and added some more specific flags.

Reply:
the 4/5 thing i really just put there to annoy people who like symmetry, sorry :)

Reply:
LOL ok ;-D

Reply:
:)...But is there any reason why we track 5 arguments for eagerness and only for 4 for "objectness"?